lib/Search/BasicSearch/BasicSearchEngine.php

```diff
@@ -0,0 +1,526 @@
+ * Software Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA
+ * 02110-1301 USA.
+ *
+ * You can contact SugarCRM, Inc. headquarters at 10050 North Wolfe Road,
+ * SW2-130, Cupertino, CA 95014, USA. or at email address contact@sugarcrm.com.
+ *
+ * The interactive user interfaces in modified source and object code versions
+ * of this program must display Appropriate Legal Notices, as required under
+ * Section 5 of the GNU Affero General Public License version 3.
+ *
+ * In accordance with Section 7(b) of the GNU Affero General Public License version 3,
+ * these Appropriate Legal Notices must retain the display of the "Powered by
+ * SugarCRM" logo and "Supercharged by SuiteCRM" logo. If the display of the logos is not
+ * reasonably feasible for technical reasons, the Appropriate Legal Notices must
+ * display the words "Powered by SugarCRM" and "Supercharged by SuiteCRM".
+ */
+
+/** @noinspection PhpUndefinedFieldInspection */
+
+namespace SuiteCRM\Search\BasicSearch;
+
+use BeanFactory;
+use DBManagerFactory;
+use JsonSchema\Exception\RuntimeException;
+use ListViewData;
+use LoggerManager;
+use SuiteCRM\Exception\Exception;
+use SuiteCRM\Search\SearchEngine;
+use SuiteCRM\Search\SearchQuery;
+use SuiteCRM\Search\SearchResults;
+use VardefManager;
+
+
+if (!defined('sugarEntry') || !sugarEntry) {
+    die('Not A Valid Entry Point');
+}
+
+/**
+ * Class BasicAndAodEngine
+ */
+class BasicSearchEngine extends SearchEngine
+{
+    /* path to search form */
+    public $searchFormPath = 'include/SearchForm/SearchForm2.php';
+
+    /*search form class name*/
+    public $searchFormClass = 'SearchForm';
+
+    public function __construct()
+    {
+        $this->cache_search = sugar_cached('modules/unified_search_modules.php');
+        $this->cache_display = sugar_cached('modules/unified_search_modules_display.php');
+    }
+
+    /**
+     * Search function run when user goes to Show All and runs a search again.  This outputs the search results
+     * calling upon the various listview display functions for each module searched on.
+     *
+     * @param SearchQuery $query
+     *
+     * @return SearchResults
+     * @throws Exception
+     */
+    public function search(SearchQuery $query): SearchResults
+    {
+        $modulesToSearch = $this->getSearchModules();
+
+        $start = microtime(true);
+
+        $results = $this->searchModules($modulesToSearch, $query->getSearchString());
+
+        $end = microtime(true);
+        $elapsed = $end - $start;
+
+        return new SearchResults($results['modules'], true, $elapsed, count($results['hits']));
+    }
+
+    /**
+     * @return array
+     */
+    protected function getSearchModules(): array
+    {
+        $unifiedSearchModuleDisplay = $this->getUnifiedSearchModulesDisplay();
+
+        require_once 'include/ListView/ListViewSmarty.php';
+
+        global $beanList, $current_user;
+
+        if (!empty($_REQUEST['advanced']) && $_REQUEST['advanced'] !== 'false') {
```

Comment:
Hi @Dillon-Brown 

I imagine this was here before, though some questions:

- Is the code expecting to receive things from `$_REQUEST`?
- The engine is one of the possible implementations right? it can be this one or lucene or other?
- If we have a generic engine that can have several implementation, we probably should receive everything to the parameters. So that all engines receive it.

what do you think?

Reply:
Removed, we won't be using request data here, this engine will always "BasicSearch" so it doesn't need to check for advanced.